cl/citations/filter_parentheticals.py

```diff
@@ -0,0 +1,76 @@
+_FULL_MODIFIABLE = f"(?:(?:{_MODIFABLE_TYPE} )?{_MODIFIABLE})"
+_QUOTE_MODIFICATION = r"(?:added|provided|removed|adopted|(?:in )?(?:the )original|omitted|included|deleted|eliminated|altered|modified|supplied|ours|mine|changed|(?:in|by) \S+|by \S+ court)"
+_DOCUMENT_TYPES = r"(?:opinion|continuance|order|op\.|decision|case|disposition|authority|authoritie|statement)"
+_OPINION_TYPES = r"(?:separate|supplemental|amended|majority|dissent|dissenting|concurrence|concurring|plurality|unpublished|revised|per curi.m|in.chambers|judgment|joint|principal|panel|unanimous|5-4|statement)"
+_OPINION_TYPE_MODIFICATION = r"(?:in (?:the )?(?:judgment|result)s?(?: in part)?|in result|in part|dubitante|in relevant part|(?:from|with|respecting) .{1,50})"  # TIL: https://en.wikipedia.org/wiki/Dubitante
+_FULL_OPINION_DESCRIPTOR = (
+    f"(?:{_OPINION_TYPES}(?: {_OPINION_TYPE_MODIFICATION})?)"
+)
+_REFERENTIAL = r"(?:quoting|citing|cited in|referencing)"
+_AGGREGATOR_TYPES = r"(?:collecting|reviewing|listing)"
+_HONORIFICS = r"(?:Mr.?|Mister)"
+_JUDGE_NAME = rf"(?:(?:.{{1,25}}J\.,?)|(?:{_HONORIFICS} Justice .{{1,25}})|the Court|(?:.{{1,25}}Circuit Justice),?)"
+
+_SURROUNDING_CHARS = r'[.!;,"“” ]'
+
+PARENTHETICAL_REGEX_BLACKLIST_RULES = [
+    r".n banc",  # en banc or in banc
+    f"{_JUDGE_NAME}(?: {_FULL_OPINION_DESCRIPTOR})?(?:[ ,]+(?:and )?{_FULL_OPINION_DESCRIPTOR})*",  # Scalia, J., dissenting; Roberts, C.J., concurring in the judgment, concurring in part, and dissenting in part
+    f"(?:{_DOCUMENT_TYPES} )?{_FULL_OPINION_DESCRIPTOR}",  # concurring in result
+    f"{_DOCUMENT_TYPES} of {_JUDGE_NAME}",  # opinion of Breyer, J.; opinion of Scalia and Alito, J.J.
+    f"{_OPINION_TYPES}(?: {_DOCUMENT_TYPES})?(?: {_OPINION_TYPE_MODIFICATION})?",  # plurality opinion, supplemental order
+    r"dictum|dicta",
+    r"on rehearing|denying cert(iorari)?",
+    r"simplified|cleaned up|as amended",
+    r"same|similar|contra",
+    r"(?:and )?cases cited therein",
+    r"No. \d+.?\d+",  # No. 12-345
+    r"n. \d+",  # n. 6
+    f"{_DOCUMENT_TYPES} below",  # case below
+    f"{_AGGREGATOR_TYPES} {_DOCUMENT_TYPES}s?",  # collecting cases, reviewing cases
+    f"(?:{_OPINION_TYPES} )?table(?: {_DOCUMENT_TYPES})?",
+    f"{_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}",  # internal citations omitted
+    f"{_FULL_MODIFIABLE} and {_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}",
+    f"{_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}[;,] ?{_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}",
+    f"(?:{_MODIFABLE_TYPE} )?{_MODIFIABLE}, {_MODIFIABLE}, and {_MODIFIABLE} {_QUOTE_MODIFICATION}",
+    f"{_REFERENTIAL} .*",  # citing Gonzales v. Raich, 123 U.S. 456 (2019). A tad over-inclusive but very helpful
+    r".{1,20} \d{4}",  # 2nd Cir. 2019, Third Circuit 1993
+    r".{1,25} I+",  # Gonzales II
+    r".{1,10} (?:Circuit|Cir.)",  # Tenth Circuit, 5th Cir.
+    r"here(?:in)?after .+",  # hereinafter, "Griffin II"
+    r"\S*",  # Single-word parentheticals, e.g., 'TILA'
```

Comment:
This, and perhaps some others, seem like they could be eliminated by adding a length requirement. Would it be helpful to require at least so many characters?